src/components/common/InputBasic.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import styled from "styled-components";
+import { inputType } from "types";
+import { COLORS } from "constants/color";
+
+interface InputBasicComponent {
+  type?: inputType;
+  placeholder?: string;
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onBlur?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  id?: string;
+  dataTestId?: string;
+  pattern?: string;
+  width?: string;
+  inputRef?:
+    | React.RefObject<HTMLInputElement>
+    | ((element: HTMLInputElement) => HTMLInputElement);
+  maxLength?: string | number;
```

Comment:
`type , placeholder, value, onChange, onBlur id, maxLength` 와 같은 속성들을 다 타입으로 정의해주고 있는데요..! 아래와 같이 엘리먼트 타입을 상속받아서 사용하면, html 에 내장된 네이티브 속성들은 따로 타입을 정의해주지 않아도 된답니다

 ```suggestion
interface InputBasicComponent extends React.InputHTMLAttributes<HTMLInputElement> {
  dataTestId?: string;
  pattern?: string;
  width?: string;
  inputRef?:
    | React.RefObject<HTMLInputElement>
    | ((element: HTMLInputElement) => HTMLInputElement);

```

Reply:
네이티브 속성은 상속을 이용하자! 메모...
덧붙이자면 pattern과 width(🤔너는 왜..?) 도 네이티브라서 생략 가능하네요!

- [refactor: inputBasic 인터페이스 중 네이티브 타입을 상속으로 변경](https://github.com/usageness/react-payments/pull/1/commits/d73d073c797353f45614425e72f5a7b21be31266)

Reply:
와 패턴!! 꿀팁 알아갑니다 줍줍